app/routes/template/templateAppendix.js

```diff
@@ -22,8 +23,29 @@ const updateSchema = schemaGenerator(db.models.templateAppendix, {
 // Add middleware to get template appendix
 router.use('/', async (req, res, next) => {
   try {
+    req.project = await db.models.project.findOne({
+      where:
+        {ident: req.body.projectId || null},
```

Comment:
this tries to get a project record with null id?

Reply:
@elewis2 if no project is provided use null as default, which will return no project. If I just use "req.body.projectId" it will break, as you cannot query with undefined

Reply:
it might be better to first check if req.body has a projectId property, then get the project or not depending on the result

Reply:
@elewis2 agreed, updating